### PR TITLE
Add libffi/3.3

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **libffi/3.3**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

Tested on Linux gcc, Visual Studio and mingw

### CODE REVIEW

**Comment by madebr** [comment]
@uilianries @danimtb The bot is stuck.

**Comment by danimtb** [comment]
Seems it is blocked by #456

**Comment by SSE4** [comment]
@danimtb how is it related to #456?

**Comment by danimtb** [comment]
@SSE4 probably I made a mistake with the number... can't see the relation 🤷‍♂

**Comment by danimtb** [comment]
please @madebr let's move this one forward when you have some time. Thanks!

File: recipes/libffi/3.3/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+            with self._create_auto_tools_environment(autotools):
+                autotools.install()
+            os.unlink(os.path.join(self.package_folder, "lib", "libffi.la"))
+            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+            tools.rmdir(os.path.join(self.package_folder, "share"))
+
+    def package_info(self):
+        self.cpp_info.includedirs = ["include"]
+        if self.options.shared:
+            self.cpp_info.defines += ["FFI_BUILDING"]
```
Comment (SSE4) [inline recipes/libffi/3.3/conanfile.py]: probably, it should be `FF_SHARED`/`FFI_STATIC` or something like that (like in 3.2.1)?
usually, `XXX_BUILDING` macro is defined while library itself is being built, not while it's being consumed.

Comment (madebr) [reply recipes/libffi/3.3/conanfile.py]: See `source_subfolder/include/ffi.h.in`:
```
/* Need minimal decorations for DLLs to work on Windows.  GCC has
   autoimport and autoexport.  Always mark externally visible symbols
   as dllimport for MSVC clients, even if it means an extra indirection
   when using the static version of the library.
   Besides, as a workaround, they can define FFI_BUILDING if they
   *know* they are going to link with the static library.  */
#if defined _MSC_VER
# if defined FFI_BUILDING_DLL /* Building libffi.DLL with msvcc.sh */
#  define FFI_API __declspec(dllexport)
# elif !defined FFI_BUILDING  /* Importing libffi.DLL */
#  define FFI_API __declspec(dllimport)
# else                        /* Building/linking static library */
#  define FFI_API
# endif
#else
# define FFI_API
#endif
```

So I was incorrect, `FFI_BUILDING` must be defined when doing a static build.

**Comment by uilianries** [comment]
Seems like CCI is not able to detect renamed folders

**Comment by danimtb** [comment]
Will check it

**Comment by uilianries** [comment]
hmm, seems like you need a separated PR for each action (new version and renaming).

**Comment by madebr** [comment]
@uilianries I've extracted 3.2 from this pr and opened #771.
This pull request is rebased on top of #771 so the ci will keep complaining until that one is merged.

**Comment by uilianries** [comment]
I see, thanks @madebr !

**Pull request merged**